src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  const temp = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const addKeys = action.extraData;
+
+      Object.assign(temp, addKeys);
+    }
+
+    if (action.type === 'removeProperties') {
+      const array = action['keysToRemove'];
+
+      for (const arr of array) {
+        if (Object.keys(temp).includes(arr)) {
+          delete temp[arr];
+        }
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const t in temp) {
+        delete temp[t];
+      }
```

Comment:
make more meaningful names
```suggestion
      for (const key in temp) {
        delete temp[key];
      }
```